tests/framework/behaviors/AttributeTypecastBehaviorTest.php

```diff
@@ -339,3 +391,37 @@ public function getAttributeTypecastBehavior()
         return $this->getBehavior('attributeTypecast');
     }
 }
+
+/**
+ * Test Active Record class with [[AttributeTypecastBehavior]] behavior attached with an enum field.
+ *
+ * @property StatusTypeString $status
+ */
+class ActiveRecordAttributeTypecastWithEnum extends ActiveRecord
+{
+    public function behaviors()
+    {
+        return [
+            'attributeTypecast' => [
+                'class' => AttributeTypecastBehavior::className(),
+                'attributeTypes' => [
+                    'status' => StatusTypeString::class,
+                ],
+                'typecastBeforeSave' => true,
```

Comment:
I am not sure if this is needed even, because other tests may test these different events/scenarios when the casting is triggered.

Reply:
Let's keep it. Won't hurt.